--- execute.c.orig	Thu Jan 19 09:49:18 2012
+++ execute.c	Fri Jun  8 13:50:08 2012
@@ -1087,7 +1087,8 @@ static int ih_function_exists(IH_HANDLER_PARAMS)
 	zend_str_tolower(lcname, func_name_len);
 #else
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "s", &lcname, &func_name_len) == FAILURE) {
-		return;
+		RETVAL_FALSE;
+		return (1);
 	}
 
 	/* Ignore leading "\" */
@@ -1501,7 +1502,8 @@ static int ih_getrandmax(IH_HANDLER_PARAMS)
 {
 #ifdef PHP_ATLEAST_5_3
 	if (zend_parse_parameters_none() == FAILURE) {
-		return;
+		RETVAL_FALSE;
+		return (1);
 	}
 #else
         int argc = ZEND_NUM_ARGS();
